pythonforandroid/bootstrap.py

```diff
@@ -257,6 +266,81 @@ def _unpack_aar(self, aar, arch):
             for f in so_files:
                 shprint(sh.cp, '-a', f, so_tgt_dir)
 
+    def collect_libraries_to_load(self, arch):
+        '''
+        Collect the shared libraries needed to load when apk initialises and
+        returns a list. This list will be used to create a file in our
+        distribution's assets folder which will be used by our java loading
+        mechanism to load those shared libraries at runtime.
+
+        .. warning:: the loading order matters, this cannot be arbitrary,
+            for example, if some lib has been build against our python, then
+            needs to be loaded after the python library has been loaded.
+        '''
+        info('Collect compiled shared libraries to load at runtime')
+
+        libs_to_load = []
+        # Add libs that should be loaded before bootstrap libs (here goes some
+        # special shared libs, like stl libs used at compile time or libs that
+        # the bootstrap may be dependant).
+        # Todo: All the recipes using gnustl_shared should be migrated to use
+        # c++_shared because in ndk r18 it will be the only stl available:
+        # https://developer.android.com/ndk/guides/cpp-support
+        stl_gnu_shared_recipes = (
+            'icu', 'leveldb', 'libzmq', 'protobuf_cpp',
+            # CppCompiledPythonRecipes depends on gnustl_shared so...
+            'atom', 'kiwisolver',)
+        stl_cxx_shared_recipes = ('boost',)
```

Comment:
Should this be here? That seems like something that should be set on each of these recipes directly, then scanned here from the global recipe list to compile this on-demand.

But that's just a minor nitpick and maybe this can be improved upon later, in overall, I like the changes I see!

Reply:
Yes, you are right, this shoudn't go here,  I will deal with this in the next commit many thanks!! :smile: